aspects/schema.go

```diff
@@ -211,6 +246,66 @@ func (s *StorageSchema) getUserType(ref string) (*userTypeRefParser, error) {
 	return nil, fmt.Errorf("cannot find user-defined type %q", ref)
 }
 
+type alternativesSchema struct {
+	// schemas holds schemas for the types allowed for the corresponding value.
+	schemas []Schema
+}
+
+// Validate that raw matches at least one of the schemas in the alternative list.
+func (v *alternativesSchema) Validate(raw []byte) error {
+	var errs []error
+	for _, schema := range v.schemas {
+		err := schema.Validate(raw)
+		if err == nil {
+			return nil
+		}
+
+		errs = append(errs, err)
+	}
+
+	var sb strings.Builder
+	for _, err := range errs {
+		sb.WriteString("\n\t")
+
+		if verr, ok := err.(*ValidationError); ok {
+			err = verr.Err
+
+			if len(verr.Path) != 0 {
+				sb.WriteString("...\"")
+				for i, part := range verr.Path {
+					switch v := part.(type) {
+					case string:
+						if i > 0 {
+							sb.WriteRune('.')
+						}
+
+						sb.WriteString(v)
+					case int:
+						sb.WriteString(fmt.Sprintf("[%d]", v))
+					default:
+						// can only happen due to bug
+						sb.WriteString(".<n/a>")
+					}
+				}
+				sb.WriteString("\": ")
+			}
+		}
+
+		sb.WriteString(err.Error())
+	}
+
+	return validationErrorf(sb.String())
+}
+
+func (v *alternativesSchema) parseConstraints(constraints map[string]json.RawMessage) error {
```

Comment:
Is it worth having another interface in addition to `parser` (`constraintsParser`?) that could be upgraded to in the cases that it is supported, and not have implementations be forced to add stubs like this?